double-conversion/string-to-double.cc

```diff
@@ -35,6 +35,11 @@
 #include "strtod.h"
 #include "utils.h"
 
+// Fix warning C4244: 'argument': conversion from 'const uc16' to 'char', possible loss of data
+#ifdef _MSC_VER
+ __pragma(warning(disable: 4244))
```

Comment:
do you know where the warning happens?

Reply:
I'll ask the developer who made the change.

Reply:
Friedemann says it was the diverse ```template <class Iterator>```, which I see are now all in ```double-conversion/string-to-double.cc```, where there's also a ```uc16 separator``` parameter, which get instantiated with ```Iterator``` either pointing to ```char *``` or to ```char16_t *``` (which I guess ```uc16 *``` is equivalent to) and which compare ```separator`` with ```**it```; that appears to happen only in ```Advance```, which the others call, so I guess it was fundamentally in ```Advance```.  Apparently it was hard to fix this correctly.  I see ```ConsumeSubStringImpl``` takes a ```converter``` that it uses to compare its ```**it``` against entries in a ```char``` sequence (C-string), so wonder (from a position of deep ignorance) whether similar might solve the problem for the other direction, comparing ```**it``` when ```char``` to a ```uc16```.

He says the warning is present in all MSVC versions since 2015.
If you prefer to skip this patch from the series, pending a fix on your own terms, I can readily leave it out.
(I'm already omitting the changes we make to ```#include``` paths, after all.)

Reply:
No it's fine.
Could you just add a small comment explaining why we disable the warning.
More or less, what you just did here in the comments.

Reply:
Ah, forgot the comment explaining why ... give me a moment ;^>